featurebyte/service/deploy.py

```diff
@@ -288,221 +368,395 @@ async def _get_enabled_serving_entity_ids(
-                        feature_online_enabled_map[feature.id] = feature.online_enabled
-                        updated_feature = await self._update_feature(
-                            feature_id=feature_id,
-                            feature_list=feature_list,
-                        )
-                        if updated_feature.online_enabled != feature.online_enabled:
-                            feature_models.append(feature)
-                            is_online_enabling = updated_feature.online_enabled
-
-                    # move update warehouse and backfill tiles to outside of transaction
-                    await self.online_enable_service.update_data_warehouse(
-                        updated_feature=updated_feature,
-                        online_enabled_before_update=feature.online_enabled,
-                    )
-
-                    if feature_update_progress:
-                        await feature_update_progress(
-                            int((ind + 1) / len(document.feature_ids) * 100),
-                            f"Updated {feature.name}",
-                        )
-
-                if feature_update_progress:
-                    await feature_update_progress(100, "Updated features")
-
-                if (
-                    target_deployed
-                    and FeastIntegrationSettings().FEATUREBYTE_FEAST_INTEGRATION_ENABLED
-                ):
-                    await self.feature_list_service.update_store_info(
-                        document_id=feature_list_id,
-                        features=all_feature_models,
-                    )
-
-                async with self.persistent.start_transaction():
-                    await self._update_feature_list_namespace(
-                        feature_list_namespace_id=feature_list.feature_list_namespace_id,
-                        feature_list=feature_list,
-                    )
-
-                await self._update_offline_store_feature_tables(
-                    feature_models,
-                    is_online_enabling,
-                    update_progress=get_ranged_progress_callback(update_progress, 80, 100)
-                    if update_progress
-                    else None,
-                )
+    def __init__(
+        self,
+        deployment_service: DeploymentService,
+        deploy_feature_list_management_service: DeployFeatureListManagementService,
+        deploy_feature_management_service: DeployFeatureManagementService,
+        feast_integration_service: FeastIntegrationService,
+        task_progress_updater: TaskProgressUpdater,
+    ):
+        self.deployment_service = deployment_service
+        self.feature_list_management_service = deploy_feature_list_management_service
+        self.feature_management_service = deploy_feature_management_service
+        self.feast_integration_service = feast_integration_service
+        self.task_progress_updater = task_progress_updater
```

Comment:
Rationale of this refactoring is to decompose the scope into 
* feature list management service
* feature manager service
* feast integration service

By delegating the responsibility of current `DeployService` into above 3 new services, it reduces the responsibility of this service.